### PR TITLE
feat(protocol-designer): change titles and names of start/end in timeline

### LINKED ISSUE #1930: Revise deck start and end state list items + sidebar title
As a user I may like to grasp initially that the side bar is chronological

## Background
I know we just changed one of these titles buuuut....I think one of the issues we saw in the usertest was that it's unclear what the sidebar is until you start making steps. This may be contributing to users doing things like trying to add ending ingredients in deck setup labware. 

## Acceptance Criteria
- [ ] Change sidebar title to "Protocol Timeline"
- [ ] Change "Labware & Liquids" to "Starting Deck State" in both the list item and page title
- [ ] Change "End" to "Final Deck State" in both the list item and page title
- [ ] Remove icons from both Start Deck State and End Deck State
- [ ] When the Starting Deck State step is pristine, display a notes-style row with instructions as row 1. Once labware is added this can disappear. 

### Pristine Start Deck State | Once labware is added
image

### LINKED ISSUE #1974: PD Bug: after deleting a step, substeps of subsequent steps will vanish
## Developer note:

This bug is happening in the `wellHighlightsForSteps` selector. It's improperly indexing the steps. There is a comment in the code where this is happening: `BUG! this doesn't work where there are deleted steps.`

### PR DESCRIPTION
**Pull request by IanLondon**
## overview

Closes #1930, closes #1974 (bug), cleans up some tech debt, and fixes 2 other unreported bugs

I decided it was time for a big refactor of steplist reducers/selectors. The problem: "initial deck setup" is not a step, it has no form, it doesn't really exist inside the timeline. But before, I had "initial deck setup" and "END" as kinda-pseudo-steps inside the steplist.

This PR makes copy changes specified in #1930, makes it so initial deck setup and "end" are **terminal items** and not steps.

## changelog

- fix bug with `robotStateTimeline` selector creating extra "steps" in the timeline for eager drop tip, introduced in #1706 
- fix related bug where substeps didn't generate for `changeTip: "never"` steps
- fix bug where hovering over Mix highlights wells of all labware, not just the relevant labware
- renamed a handful of selectors touched by this PR to `getSpam` instead of `spam`
- create TerminalItem component
- change copy of text in initial deck setup, final deck state, and corresponding headers
- remove flask icon from initial deck setup
- remove `StepTypeWithEnd` and remove `any` typing that was a workaround for it
- remove `deckSetupMode` selector
- update all places timeline index is used to no longer add 1 to the index to skip over deck setup step (timeline warnings, timeline errors, liquids on deck at step in timeline, well selection)

## review requests

:warning: Breaking change for loading files, old files will not work :warning: 

Please test extensively, this touches all parts of PD that have to do with steps in the timeline:
- Steps
- Warnings
- Errors
- Liquid tracking
- Substeps
- Saving and loading (don't try a file from an older version, it won't work right)

### CODE REVIEW

File: protocol-designer/src/steplist/reducers.js [hunk 1]
```diff
@@ -90,16 +95,10 @@ const unsavedFormModal = handleActions({
...
-    id: INITIAL_DECK_SETUP_ID,
-    title: DECK_SETUP_TITLE,
-    stepType: 'deck-setup'
-  }
-}
+const initialStepState = {}
 
-const steps = handleActions({
-  ADD_STEP: (state, action: AddStepAction) => ({
+const steps: Reducer<StepsState, *> = handleActions({
```
Comment (IanLondon) [inline protocol-designer/src/steplist/reducers.js]: we should start typing reducers this way with `Reducer` type from `redux-actions`! The way they're typed now isn't very safe

File: protocol-designer/src/containers/SelectablePlate.js [hunk 2]
```diff
@@ -64,42 +67,60 @@ function mapStateToProps (state: BaseState, ownProps: OP): SP {
...
   } else {
-    // well contents for step, not inital state.
-    // shows liquids the current step in timeline
-    const wellContentsWithoutHighlight = (stepId === END_STEP || !allWellContentsForSteps[prevStepId])
-      // End step, or erroring step: show last valid well contents in timeline
-      ? wellContentsSelectors.lastValidWellContents(state)[containerId]
-      // Valid non-end step
-      : allWellContentsForSteps[prevStepId][containerId]
+    const stepId = activeItem.id
+    // TODO: Ian 2018-07-31 replace with util function, "findIndexOrNull"?
```
Comment (IanLondon) [inline protocol-designer/src/containers/SelectablePlate.js]: thoughts? I hate that `findIndex` (even in lodash) returns `-1`

File: protocol-designer/src/file-data/selectors/commands.js [hunk 3]
```diff
@@ -192,8 +195,7 @@ export const timelineWarningsPerStep: Selector<WarningsPerStep> = createSelector
   steplistSelectors.orderedSteps,
   robotStateTimeline,
   (orderedSteps, timeline) => timeline.timeline.reduce((acc: WarningsPerStep, frame, timelineIndex) => {
-    // TODO: Ian 2018-06-15 add 1 to orderedSteps because 0th orderedStep is deck setup. DRYer way?
-    const stepId = orderedSteps[timelineIndex + 1]
```
Comment (IanLondon) [inline protocol-designer/src/file-data/selectors/commands.js]: phew

File: protocol-designer/src/components/steplist/TerminalItem/index.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  hovered: $ElementType<Props, 'hovered'>,
+  selected: $ElementType<Props, 'selected'>
+}
+
+function mapStateToProps (state: BaseState, ownProps: OP): SP {
+  const {id} = ownProps
+  const hovered = steplistSelectors.getHoveredTerminalItemId(state) === id
+  const selected = steplistSelectors.getSelectedTerminalItemId(state) === id
+  return {
+    hovered,
```
[the remark this answers is not in the document]
Comment (IanLondon) [reply protocol-designer/src/components/steplist/TerminalItem/index.js]: nope they're `?boolean`s

[the remark this answers is not in the document]
Comment (b-cooper) [reply protocol-designer/src/components/steplist/TerminalItem/index.js]: aha, then I'd vote for `isHovered` & `isSelected`

File: protocol-designer/src/steplist/actions/actions.js [hunk 5]
```diff
@@ -101,51 +98,56 @@ export const toggleStepCollapsed = (stepId: StepIdType): ToggleStepCollapsedActi
...
   payload: payload
 })
 
-export const selectStep = (stepId: StepIdType | EndStepId): ThunkAction<*> =>
+export type SelectTerminalItemAction = {
+  type: 'SELECT_TERMINAL_ITEM',
+  payload: TerminalItemId
+}
+
+const selectTerminalItemAction = (terminalId): SelectTerminalItemAction => ({
```
[the remark this answers is not in the document]
Comment (IanLondon) [reply protocol-designer/src/steplist/actions/actions.js]: it's kinda the same as `selectStepAction`, a "private" non-exported action creator that's only called by a thunk. But now that I check back, `selectedStepAction` is actually the action object, not a fn, so I am follow that pattern more closely for `selectTerminalItemAction`

File: protocol-designer/src/components/steplist/StepList.js [hunk 6]
```diff
@@ -2,29 +2,34 @@
...
 import {SidePanel} from '@opentrons/components'
 
-import {END_STEP, type StepIdTypeWithEnd} from '../../steplist/types'
-import type {StepIdType} from '../../form-types'
-
+import StartingDeckStateTerminalItem from './StartingDeckStateTerminalItem'
 import StepItem from '../../containers/ConnectedStepItem'
 import StepCreationButton from '../../containers/StepCreationButton'
+import TerminalItem from './TerminalItem'
+import {END_TERMINAL_TITLE} from '../../constants'
```
[the remark this answers is not in the document]
Comment (IanLondon) [reply protocol-designer/src/components/steplist/StepList.js]: good point, I'll add "item". I have the titles separate b/c they're used by the steplist but also the ConnectedTitleBar so they don't belong in `steplist/types.js` -- I think they'll move into the string JSON soon enough

**Pull request merged**